lib/rmt/cli/array_printable.rb

```diff
@@ -1,12 +1,25 @@
-module RMT::CLI::ArrayPrintable
+require 'csv'
+require 'terminal-table'
 
-  def array_to_table(array, options)
-    rows = []
+module RMT::CLI::ArrayPrintable
 
-    array.all.each do |element|
-      rows << options.keys.map { |k| element[k] }
+  def format_array(array, array_options)
+    rows = array.map do |element|
+      array_options.keys.map { |k| element.public_send(k) }
     end
-    Terminal::Table.new headings: options.values, rows: rows
+    if options.csv
```

Comment:
I don't like that this module method has a hidden dependency on the Thor gem's options object. This should be part of the method definition.

Reply:
Agreed.